### PR TITLE
Added tests for bootman/timeout.c

### PR DESCRIPTION
**Pull request by daniel-schuermann**

### CODE REVIEW

**Comment by ikeydoherty** [comment]
Odd. Travis logs should a failure:

```
make[2]: Entering directory `/home/travis/build/ikeydoherty/clr-boot-manager'
make[3]: Entering directory `/home/travis/build/ikeydoherty/clr-boot-manager'
FAIL: check_core
PASS: check_files
PASS: check_updates
============================================
1 of 3 tests failed
See ./test-suite-memcheck.log
Please report to michael.i.doherty@intel.com
============================================
make[3]: *** [test-suite-memcheck.log] Error 1
make[3]: Leaving directory `/home/travis/build/ikeydoherty/clr-boot-manager'
make[2]: *** [check-TESTS] Error 2
```

**Comment by ikeydoherty** [comment]
OK so your PR is very good in content, thanks! Singular nit-pick: The commit message should begin with an uppercase A.

You also exposed a memory leak with this test too, which I guess is all part and parcel of it:

```
==14062== 83 bytes in 1 blocks are definitely lost in loss record 47 of 53
==14062==    at 0x4C2F342: realloc (in /usr/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==14062==    by 0x5C1839F: __vasprintf_chk (vasprintf_chk.c:88)
==14062==    by 0x5C182A1: __asprintf_chk (asprintf_chk.c:32)
==14062==    by 0x4099CA: asprintf (stdio2.h:178)
==14062==    by 0x4099CA: boot_manager_set_timeout_value (timeout.c:37)
==14062==    by 0x402D6E: bootman_timeout_test (check-core.c:517)
==14062==    by 0x54C62B7: srunner_run (in /usr/lib64/libcheck.so.0.0.0)
==14062==    by 0x402B02: main (check-core.c:568)
```

TL;DR The `char *path` should be an autofree variable. It's up to you if you want your revised patch to include a commit for that or not  :)

**Comment by bryteise** [comment]
@ikeydoherty So that's a little tough for me to understand what is going on from the raw log. It runs make check in some form 5 times and one of them does look to fail on one of the tests, maybe in make distcheck? I'll take a look at it in a few.

**Comment by daniel-schuermann** [comment]
Thanks for your appreciation! I can include the patch. Do you prefer a single commit?
Small additional question: where do I find the test of the main function? It's marked as tested, but I don't find it.

-------- Originalnachricht --------
Betreff: Re: [ikeydoherty/clr-boot-manager] Added tests for bootman/timeout.c (#20)
Von: Ikey Doherty
An: ikeydoherty/clr-boot-manager
Cc: "Schürmann, Daniel" ,Author


OK so your PR is very good in content, thanks! Singular nit-pick: The commit message should begin with an uppercase A.

You also exposed a memory leak with this test too, which I guess is all part and parcel of it:

==14062== 83 bytes in 1 blocks are definitely lost in loss record 47 of 53
==14062==    at 0x4C2F342: realloc (in /usr/lib64/valgrind/vgpreload_memcheck-amd64-linux.so)
==14062==    by 0x5C1839F: __vasprintf_chk (vasprintf_chk.c:88)
==14062==    by 0x5C182A1: __asprintf_chk (asprintf_chk.c:32)
==14062==    by 0x4099CA: asprintf (stdio2.h:178)
==14062==    by 0x4099CA: boot_manager_set_timeout_value (timeout.c:37)
==14062==    by 0x402D6E: bootman_timeout_test (check-core.c:517)
==14062==    by 0x54C62B7: srunner_run (in /usr/lib64/libcheck.so.0.0.0)
==14062==    by 0x402B02: main (check-core.c:568)


TL;DR The char *path should be an autofree variable. It's up to you if you want your revised patch to include a commit for that or not :)

-

**Comment by ikeydoherty** [comment]
@bryteise its `make check-valgrind` that fails

@daniel-schuermann whatever works best for you - its all part of the same problem :D 

The bulk of the tests for CBM are actually in `check-updates.c`

**Comment by ikeydoherty** [comment]
Going to merge in the current form, thanks for the change :) Increasing coverage is an important task for CBM!
